clientUtils/search.tsx

```diff
@@ -0,0 +1,207 @@
+    for (const replacement of moreTolerantMatchReplacements) {
+        moreTolerantMatch = moreTolerantMatch.replace(
+            replacement.searchTerm,
+            replacement.regexTerm
+        )
+    }
+    return new RegExp(moreTolerantMatch, "iu")
+}
+export const buildSearchWordsFromSearchString = (
+    searchInput: string | undefined
+): SearchWord[] => {
+    if (!searchInput) return []
+    const wordRegexes = searchInput
+        .split(" ")
+        .filter((item) => item)
+        .map((item) => ({
+            regex: buildRegexFromSearchWord(item),
+            word: item,
+        }))
+    return wordRegexes
+}
+
+/** Given a list of SearchWords constructed with buildSearchWordsFromSearchString
+    and a search field string extractor function, this function returns a filter function
+    that tells you if all search terms occur in any of the fields extracted by the extractor fn.
+
+    E.g. if you have a type Person with firstName and lastName and you want it to search in both fields,
+    you would call it like this:
+
+    @example
+    const searchWords = buildSearchWordsFromSearchString("peter mary")
+    const filterFn = filterFunctionForSearchWords(searchWords, (person: Person) => [person.firstName, person.lastName])
+    const filteredPeople = people.filter(filterFn)
+*/
+export function filterFunctionForSearchWords<TargetObject>(
+    searchWords: SearchWord[],
+    targetPropertyExtractorFn: (t: TargetObject) => (string | undefined)[]
+): (target: TargetObject) => boolean {
+    return (target: TargetObject): boolean =>
+        searchWords.every((searchWord) =>
+            targetPropertyExtractorFn(target).some(
+                (searchString) =>
+                    searchString !== undefined &&
+                    searchWord.regex.test(searchString)
+            )
+        )
+}
+
+export function highlightFunctionForSearchWords(
+    searchWords: SearchWord[]
+): (text: string) => JSX.Element | string {
+    return (text: string): JSX.Element | string => {
+        if (searchWords.length > 0) {
+            const firstMatches = searchWords
+                .map((regex) => [text.search(regex.regex), regex.word] as const)
+                .filter(([index, word]) => index >= 0)
+            const sortedFirstMatches = sortBy(
+                firstMatches,
+                ([index, word]) => index
+            )
+            const fragments: JSX.Element[] = []
+            let lastIndex = 0
+            for (const [index, word] of sortedFirstMatches) {
+                fragments.push(
+                    <span key={`${lastIndex}-start`}>
+                        {text.substring(lastIndex, index)}
+                    </span>
+                )
+                fragments.push(
+                    <span
+                        key={`${lastIndex}-content`}
+                        style={{ color: "#aa3333" }}
+                    >
+                        {word}
```

Comment:
I feel like it would be better to use the original casing here, as it is in the searchString.
Consider this for example:
![image](https://user-images.githubusercontent.com/2641501/131398937-398ea38f-0ee6-45a2-a76b-1535448d9d2e.png)

Imho it should be "Output-side real ***GD***P per capita"